src/platform/windows.c

```diff
@@ -1126,7 +1117,7 @@ static bool windows_setup_file_association(
 	}
 	// [hRootKey\OpenRCT2.ext\DefaultIcon]
 	wchar_t szIconW[MAX_PATH];
-	wsprintfW(szIconW, L"\"%s\",%d", dllPathW, iconIndex);
+	StringCchPrintfW(szIconW, MAX_PATH, L"\"%s\",%d", dllPathW, iconIndex);
```

Comment:
*facepalm* Didn't realize that was a thing 😜. It does seem that it's slightly different, though, as it handles truncation rather differently (though probably in a more preferable way). Don't know if it's in MinGW, though, so I'll have to try a build and see.